modules/indexer/issues/meilisearch/meilisearch.go

```diff
@@ -223,6 +226,31 @@ func (b *Indexer) Search(ctx context.Context, options *internal.SearchOptions) (
 
 	hits := make([]internal.Match, 0, len(searchRes.Hits))
 	for _, hit := range searchRes.Hits {
+		if !options.IsFuzzyKeyword {
+			// as meilisearch does not have a non-fuzzy search and you can only change "typo tolerance" per index we have to post-filter the results
+			// https://www.meilisearch.com/docs/learn/configuration/typo_tolerance#configuring-typo-tolerance
+			// TODO: remove once https://github.com/orgs/meilisearch/discussions/377 is addressed
+			keyword := strings.ToLower(options.Keyword)
+			title, _ := hit.(map[string]any)["title"].(string)
+			if !strings.Contains(strings.ToLower(title), keyword) {
+				content, _ := hit.(map[string]any)["content"].(string)
+				if !strings.Contains(strings.ToLower(content), keyword) {
+					comments, _ := hit.(map[string]any)["comments"].([]any)
+					found := false
+					for i := range comments {
+						comment, _ := comments[i].(string)
+						if strings.Contains(strings.ToLower(comment), keyword) {
+							found = true
+							break
+						}
+					}
+					if !found {
+						// we could not have a direct match, so ignore that hit and move on ...
+						continue
+					}
+				}
+			}
+		}
```

Comment:
Yes, extract pure function and unit-test it please. Also throw weird input at it.

Reply:
why should i throw weird json into it, in this case it just wont do anything ... beside later in the code it would panic, I think I'll catch that in this pull too

Reply:
Yeah I was just recommending what I would do in such unit tests. If it's already safe, just test at least one positive and one negative case.